File: src/main/kotlin/inflearn/java/_1_4_단어_뒤집기.java
package inflearn.java;

import java.util.ArrayList;
import java.util.List;
import java.util.Scanner;

public class _1_4_단어_뒤집기 {

    public static List<String> solution(List<String> input) {
        List<String> answer = new ArrayList<String>();

        for(String str: input) {
            answer.add((new StringBuilder(str)).reverse().toString());
        }

        return answer;
    }

    public static List<String> solution2(List<String> input) {
        ArrayList<String> answer = new ArrayList<>();
        for(String x : input) {
            char[] s = x.toCharArray();
            int lt = 0, rt = x.length() - 1;
            while(lt < rt) {
                char tmp = s[lt];
                s[lt] = s[rt];
                s[rt] = tmp;
                lt++;
                rt--;
            }
            String tmp = String.valueOf(s);
            answer.add(tmp);
        }

        return answer;
    }

    public static void main(String[] args) {
        Scanner sc = new Scanner(System.in);
        int n = sc.nextInt();
        List<String> input = new ArrayList<String>();
        for(int i = 0; i < n; i++) {
            input.add(sc.next());
        }
        for(String str: solution(input)) {
            System.out.println(str);
        }
    }
}

File: src/main/kotlin/inflearn/kotlin/ch01/1_4 단어 뒤집기.kt
package inflearn.kotlin.ch01

import java.util.*

object `1_4 단어 뒤집기` {
    fun solution(input: List<String>): List<String> {
        val answer = mutableListOf<String>()
        for (str in input) {
            answer.add(StringBuilder(str).reverse().toString())
        }
        return answer
    }

    @JvmStatic
    fun main(args: Array<String>) {
        val sc = Scanner(System.`in`)
        val n = sc.nextInt()
        val input: MutableList<String> = ArrayList()
        for (i in 0 until n) {
            input.add(sc.next())
        }
        for (str in solution(input)) {
            println(str)
        }
    }
}